specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,280 @@
+/// and allows the host to provide authentication response or cancel the request.
+[uuid(51d3adaa-159f-4e48-ad39-a86beb2c1435), object, pointer_default(unique)]
+interface ICoreWebView2WebAuthenticationRequestedEventArgs : IUnknown
+{
+  /// The web resource request that led to the authorization challenge
+  [propget] HRESULT Request([out, retval] ICoreWebView2WebResourceRequest** request);
+
+  /// The authorization challenge string
+  [propget] HRESULT Challenge([out, retval] LPWSTR* challenge);
+
+  /// Response to the authentication request with credentials.
+  [propget] HRESULT Response([out, retval] ICoreWebView2WebAuthenticationResponse** response);
+  /// Set the Response property.
+  [propput] HRESULT Response([in] ICoreWebView2WebAuthenticationResponse* response);
+
+  /// Cancel the authentication request. False by default.
+  /// If set to true, Response will be ignored.
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+  /// Set the Cancel property.
+  [propput] HRESULT Cancel([in] BOOL cancel);
+}
+
+[uuid(0cec3e32-36aa-4859-9bbe-f9c116ad4721), object, pointer_default(unique)]
+interface ICoreWebView2Environment : IUnknown
+{
+  /// Create a WebAuthenticationResponse object used to provide credentials for
+  /// WebAuthenticationRequested event
+  HRESULT CreateWebAuthenticationResponse(
+    [in] LPCWSTR userName,
+    [in] LPCWSTR password,
+    [out, retval] ICoreWebView2WebAuthenticationResponse** response);
+}
+
+```
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Environment
+    {
+        ...
+
+        /// Create a WebAuthenticationResponse object used to provide credentials for
+        /// WebAuthenticationRequested event
+        public CoreWebView2WebAuthenticationResponse CreateWebAuthenticationResponse(
+            String userName,
+            String password);
+    }
+
+    /// Event args for the WebAuthorizationRequested event. Will contain the
+    /// request that led to the HTTP authorization challenge, the challenge
+    /// and allows the host to provide authentication response or cancel the request.
+    runtimeclass CoreWebViewWebAuthenticationRequestedEventArgs
+    {
+        /// The web resource request that led to the authorization challenge
+        CoreWebView2WebResourceRequest Request { get; };
+
+        /// The HTTP basic authorization challenge string
+        String Challenge { get; };
+
+        /// Cancel the authentication request. False by default.
+        /// If set to true, Response will be ignored.
+        bool Cancel { get; set; };
+
+        /// Response to the authentication request with credentials.
+        CoreWebView2WebAuthenticationResponse Response { get; set; };
+    }
+
+    /// Represents a Basic HTTP authentication response that contains a user name
+    /// and a password as according to RFC7617 (https://tools.ietf.org/html/rfc7617)
+    runtimeclass CoreWebView2WebAuthenticationResponse
```

Comment:
Do we need a separate response class? Versus just having username and password properties on the event args?

Reply:
A separate object fits in to the pattern we have with WebResourceRequested events.

Reply:
But it seems like unnecessary additional work for the end dev. 

I'm looking at the existing sample code for setting the username and password:
```
    CoreWebView2WebAuthenticationResponse response =
        _coreWebView2Environment.CreateWebAuthenticationResponse(
            credential.username, credential.password);
    args.Response = response;
```

It seems like we shouldn't need CreateWebAuthenticationResponse and it would be easier for the end dev to figure out what code to write. They don't have to go find Create on the environment object. The Response property on the event args could be just a getter property and the username and password is mutable

```
args.Response.Username = credential.username;
args.Response.Password = credential.password;
```

Or we put the username and password directly on the event args as mutable properties:
```
args.Username = credential.username;
args.Password = credential.password;
```